File: app/src/main/java/me/tatarka/betterfragment/sample/LifecycleLogger.java
package me.tatarka.betterfragment.sample;

import android.util.Log;

import javax.inject.Inject;
import javax.inject.Singleton;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.OnLifecycleEvent;

@Singleton
public class LifecycleLogger implements LifecycleObserver {

    @Inject
    public LifecycleLogger() {
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    public void onCreate(LifecycleOwner owner, Lifecycle.Event event) {
        Log.d("LIFECYCLE", event + " " + owner);
    }
}

File: app/src/main/java/me/tatarka/shard/sample/LifecycleLogger.kt
package me.tatarka.shard.sample

import android.util.Log

import javax.inject.Inject
import javax.inject.Singleton

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.OnLifecycleEvent

@Singleton
class LifecycleLogger @Inject constructor() : LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
    fun onCreate(owner: LifecycleOwner, event: Lifecycle.Event) {
        Log.d("LIFECYCLE", "$event $owner")
    }
}
